axmol compilation fails with C++23 - it looks like real bug in the code

- axmol version: latest 
- devices test on: all
- developing environments
   - NDK version: r23c
   - Xcode version: 14.2+
   - Visual Studio: 
      - VS version: 2022 (17.9+)
      - MSVC version: 19.39+
      - Windows SDK version: 10.0.22621.0+
   - cmake version: 
Steps to Reproduce:

branch: dev
https://github.com/axmolengine/axmol/pull/2414
branch: v3
https://github.com/axmolengine/axmol/pull/2413

Sample error:
```
C/C++: /home/runner/work/axmol/axmol/core/base/Logging.cpp:160:12: error: non-const lvalue reference to type 'LogItem' cannot bind to a temporary of type 'LogItem'
C/C++:   160 |     return item;
```

--- a/cmake/Modules/AXConfigDefine.cmake
+++ b/cmake/Modules/AXConfigDefine.cmake
@@ -64,6 +64,8 @@ if (CMAKE_CXX_STANDARD GREATER_EQUAL ${_AX_MIN_CXX_STD})
 else()
     message(STATUS "Building axmol require c++ std >= ${_AX_MIN_CXX_STD}")
 endif()
+
+# used to set 3rdparty c++ standard same with axmol
 set(_AX_CXX_STD ${CMAKE_CXX_STANDARD} CACHE STRING "" FORCE)
 
 if(NOT DEFINED CMAKE_CXX_STANDARD_REQUIRED)

--- a/core/base/Logging.cpp
+++ b/core/base/Logging.cpp
@@ -68,7 +68,7 @@ AX_API void setLogOutput(ILogOutput* output)
     s_logOutput = output;
 }
 
-AX_API LogItem& preprocessLog(LogItem&& item)
+AX_API LogItem&& preprocessLog(LogItem&& item)
 {
     if (s_logFmtFlags != LogFmtFlag::Null)
     {
@@ -147,8 +147,7 @@ AX_API LogItem& preprocessLog(LogItem&& item)
             prefix_size += fmt::format_to_n(wptr + prefix_size, buffer_size - prefix_size,
                                             "[{}-{:02d}-{:02d} {:02d}:{:02d}:{:02d}.{:03d}]", ts.tm_year + 1900,
                                             ts.tm_mon + 1, ts.tm_mday, ts.tm_hour, ts.tm_min, ts.tm_sec,
-                                            static_cast<int>(tv_msec % std::milli::den))
-                               .size;
+                                            static_cast<int>(tv_msec % std::milli::den)).size;
         }
         if (bitmask::any(s_logFmtFlags, LogFmtFlag::ProcessId))
             prefix_size +=
@@ -157,13 +156,15 @@ AX_API LogItem& preprocessLog(LogItem&& item)
             prefix_size +=
                 fmt::format_to_n(wptr + prefix_size, buffer_size - prefix_size, "[TID:{:x}]", xmol_gettid()).size;
     }
-    return item;
+    return std::forward<LogItem>(item);
 }
 
-AX_DLL void outputLog(LogItem& item, const char* tag)
+AX_DLL void outputLog(LogItem&& item, const char* tag)
 {
-    if (!s_logOutput) writeLog(item, tag);
-    else s_logOutput->write(item, tag);
+    if (!s_logOutput)
+        writeLog(item, tag);
+    else
+        s_logOutput->write(item, tag);
 }
 
 AX_DLL void writeLog(LogItem& item, const char* tag)
@@ -210,7 +211,7 @@ AX_DLL void writeLog(LogItem& item, const char* tag)
         OutputDebugStringW(ntcvt::from_chars(item.message()).c_str());
 #    endif
 
-        // write normal color text to console
+    // write normal color text to console
 #    if defined(_WIN32)
     auto hStdout = ::GetStdHandle(item.level_ != LogLevel::Error ? STD_OUTPUT_HANDLE : STD_ERROR_HANDLE);
     if (hStdout)
@@ -244,4 +245,4 @@ AX_API void print(const char* format, ...)
                   "axmol debug info");
 }
 #endif
-}
+}  // namespace ax

--- a/core/base/Logging.h
+++ b/core/base/Logging.h
@@ -57,7 +57,7 @@ AX_ENABLE_BITMASK_OPS(LogFmtFlag);
 
 class LogItem
 {
-    friend AX_API LogItem& preprocessLog(LogItem&& logItem);
+    friend AX_API LogItem&& preprocessLog(LogItem&& logItem);
     friend AX_API void writeLog(LogItem& item, const char* tag);
 
 public:
@@ -77,7 +77,7 @@ public:
     }
 
     template <typename _FmtType, typename... _Types>
-    inline static LogItem& vformat(_FmtType&& fmt, LogItem& item, _Types&&... args)
+    inline static LogItem&& vformat(_FmtType&& fmt, LogItem&& item, _Types&&... args)
     {
         item.qualified_message_ =
             fmt::format(std::forward<_FmtType>(fmt), std::string_view{item.prefix_buffer_, item.prefix_size_},
@@ -89,7 +89,7 @@ public:
         if (item.has_style_)
             item.qualified_message_.append("\x1b[m"sv);
         item.qualifier_size_ += (item.qualified_message_.size() - old_size);
-        return item;
+        return std::forward<LogItem>(item);
     }
 
 private:
@@ -124,17 +124,19 @@ AX_API void setLogFmtFlag(LogFmtFlag flags);
 AX_API void setLogOutput(ILogOutput* output);
 
 /* @brief internal use */
-AX_API LogItem& preprocessLog(LogItem&& logItem);
+AX_API LogItem&& preprocessLog(LogItem&& logItem);
 
 /* @brief internal use */
-AX_API void outputLog(LogItem& item, const char* tag);
+AX_API void outputLog(LogItem&& item, const char* tag);
 AX_API void writeLog(LogItem& item, const char* tag);
 
 template <typename _FmtType, typename... _Types>
-inline void printLogT(_FmtType&& fmt, LogItem& item, _Types&&... args)
+inline void printLogT(_FmtType&& fmt, LogItem&& item, _Types&&... args)
 {
     if (item.level() >= getLogLevel())
-        outputLog(LogItem::vformat(std::forward<_FmtType>(fmt), item, std::forward<_Types>(args)...), "axmol");
+        outputLog(
+            LogItem::vformat(std::forward<_FmtType>(fmt), std::forward<LogItem>(item), std::forward<_Types>(args)...),
+            "axmol");
 }
 
 #define AXLOG_WITH_LEVEL(level, fmtOrMsg, ...) \
@@ -158,7 +160,7 @@ inline void printLogT(_FmtType&& fmt, LogItem& item, _Types&&... args)
 #define AXLOGW(fmtOrMsg, ...) AXLOG_WITH_LEVEL(ax::LogLevel::Warn, fmtOrMsg, ##__VA_ARGS__)
 #define AXLOGE(fmtOrMsg, ...) AXLOG_WITH_LEVEL(ax::LogLevel::Error, fmtOrMsg, ##__VA_ARGS__)
 
-#define AXLOGT AXLOGV
+#define AXLOGT                AXLOGV
 
 #ifndef AX_CORE_PROFILE
 /**
@@ -166,4 +168,4 @@ inline void printLogT(_FmtType&& fmt, LogItem& item, _Types&&... args)
  */
 /* AX_DEPRECATED(2.1)*/ AX_API void print(const char* format, ...) AX_FORMAT_PRINTF(1, 2);  // use AXLOGD instead
 #endif
-}
+}  // namespace ax

